### PR TITLE
Home details and report

### PR DESCRIPTION
**Pull request by shailee-m**
Adding details and report page for each home.
Depends on #5

### CODE REVIEW

File: frontend/src/features/home-details/services/detail-services.js [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+  };
+
+  // Get client locale
+  const locale = i18n.locale;
+
+  // Render plot
+  Plotly.newPlot("homeActivityCountsByFacilitatorRoleChart", data, layout, {
+    locale
+  });
+};
```
Comment (brylie) [inline frontend/src/features/home-details/services/detail-services.js]: Just for clarity of organization, could we please move the chart rendering functions into a different module? E.g. `features/home-details/charts/charts.js`? That way, the current file would deal *only* with data fetching, while the `charts.js` file would deal only with chart rendering. Kind of like separating forks and knives.

File: frontend/src/features/home-details/services/detail-services.js [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+    },
+    bargap: 0.05,
+    showlegend: false
+  };
+
+  // Get client locale
+  const locale = i18n.locale;
+
+  // Render plot
+  Plotly.newPlot("homeActivityCountsByFacilitatorRoleChart", data, layout, {
```
Comment (brylie) [inline frontend/src/features/home-details/services/detail-services.js]: I think the DOM element selector should be passed in as an argument to the render function. That way, it is apparent in the HTML template file which element is responsible for containing the chart. Then if the HTML element is changed, it is just a matter of changing an argument to the chart render function in the same file, as opposed to having to locate this line of code in a separate file.

**Review by brylie — CHANGES_REQUESTED** [review]

**Review by brylie — CHANGES_REQUESTED** [review]
Make sure to run automatic code formatting on the Vue template files. I recommend using the Vetur VS Code plugin, as part of this extension pack:

https://marketplace.visualstudio.com/items?itemName=sdras.vue-vscode-extensionpack

**Comment by brylie** [comment]
When rendering an individual Home page, The following error appears in a toast message:

```
TypeError: d3.nest is not a function<br> &nbsp; &nbsp;at Object.getHomeActivityTypeMetrics (server/methods/charts/homeCharts.js:125:15)<br> &nbsp; &nbsp;at packages/simple_rest.js:383:33<br> &nbsp; &nbsp;at packages/simple_json-routes.js:98:9
```

**Review by brylie — CHANGES_REQUESTED** [review]
We need to include `package-lock.json` in our repository.

**Comment by shailee-m** [comment]
> We need to include `package-lock.json` in our repository.

the project was setup with yarn by default. yarn.lock is committed. Should I remove yarn?

**Comment by shailee-m** [comment]
> ```
> TypeError: d3.nest is not a function<br> &nbsp; &nbsp;at Object.getHomeActivityTypeMetrics (server/methods/charts/homeCharts.js:125:15)<br> &nbsp; &nbsp;at packages/simple_rest.js:383:33<br> &nbsp; &nbsp;at packages/simple_json-routes.js:98:9
> ```

I think the yarn.lock has the exact d3 version. 
Can your reinstall the node_modules using the command `yarn` and check?

**Comment by brylie** [comment]
Oops. I forgot we are using Yarn here. My bad.

**Review by brylie — APPROVED** [review]

**Pull request merged**